File: app/src/main/java/com/botijasoftware/utils/ColorRGBb.java
package com.botijasoftware.utils;

public class ColorRGBb {
	
	public ColorRGBb() {}

	public ColorRGBb(byte c) {
		R = c;
		G = c;
		B = c;
	}

	public ColorRGBb(byte r, byte g, byte b) {
		R = r;
		G = g;
		B = b;
	}

	public void setValue(byte c) {
		R = c;
		G = c;
		B = c;
	}

	public void setValue(byte r, byte g, byte b) {
		R = r;
		G = g;
		B = b;
	}


	public ColorRGBb clone() {
		return new ColorRGBb(R,G,B);
	}
	
	public void saturate() {
		/*float m = (R + G + B / 3.0f);
		if (m != 0.0f) {
			m = 1.0f/m;
			R = (byte) (R * m);
			G = (byte) (G * m);
			B = (byte) (B * m);
		}*/
	}
	
	public byte R;
	public byte G;
	public byte B;
	
	public static final ColorRGBb BLACK = new ColorRGBb((byte)0, (byte)0, (byte)0);
	public static final ColorRGBb WHITE = new ColorRGBb((byte)255, (byte)255, (byte)255);
	public static final ColorRGBb RED = new ColorRGBb((byte)255,(byte)0,(byte)0);
	public static final ColorRGBb LIME = new ColorRGBb((byte)0,(byte)255,(byte)0);
	public static final ColorRGBb BLUE = new ColorRGBb((byte)0,(byte)0,(byte)255);
	public static final ColorRGBb YELLOW = new ColorRGBb((byte)255,(byte)255,(byte)0);
	public static final ColorRGBb CYAN = new ColorRGBb((byte)0,(byte)255,(byte)255);
	public static final ColorRGBb MAGENTA = new ColorRGBb((byte)255,(byte)0,(byte)255);
	public static final ColorRGBb SILVER = new ColorRGBb((byte)192,(byte)192,(byte)192);
	public static final ColorRGBb GRAY = new ColorRGBb((byte)128,(byte)128,(byte)128);
	public static final ColorRGBb MAROON = new ColorRGBb((byte)128,(byte)0,(byte)0);
	public static final ColorRGBb OLIVE = new ColorRGBb((byte)128,(byte)128,(byte)0);
	public static final ColorRGBb GREEN = new ColorRGBb((byte)0,(byte)128,(byte)0);
	public static final ColorRGBb PURPLE = new ColorRGBb((byte)128,(byte)0,(byte)128);
	public static final ColorRGBb TEAL = new ColorRGBb((byte)0,(byte)128,(byte)128);
	public static final ColorRGBb NAVY = new ColorRGBb((byte)0,(byte)0,(byte)128);
	
}

File: app/src/main/java/com/botijasoftware/utils/ColorRGBb.kt
package com.botijasoftware.utils

class ColorRGBb {

    constructor() {
        B = 0
        G = B
        R = G
    }

    constructor(c: Byte) {
        R = c
        G = c
        B = c
    }

    constructor(r: Byte, g: Byte, b: Byte) {
        R = r
        G = g
        B = b
    }

    constructor(c: ColorRGBb) {
        R = c.R
        G = c.G
        B = c.B
    }

    fun setValue(c: Byte) {
        R = c
        G = c
        B = c
    }

    fun setValue(r: Byte, g: Byte, b: Byte) {
        R = r
        G = g
        B = b
    }


    fun clone(): ColorRGBb {
        return ColorRGBb(R, G, B)
    }

    fun saturate() {
        /*float m = (R + G + B / 3.0f);
		if (m != 0.0f) {
			m = 1.0f/m;
			R = (byte) (R * m);
			G = (byte) (G * m);
			B = (byte) (B * m);
		}*/
    }

    var R: Byte = 0
    var G: Byte = 0
    var B: Byte = 0

    companion object {

        val BLACK = ColorRGBb(0.toByte(), 0.toByte(), 0.toByte())
        val WHITE = ColorRGBb(255.toByte(), 255.toByte(), 255.toByte())
        val RED = ColorRGBb(255.toByte(), 0.toByte(), 0.toByte())
        val LIME = ColorRGBb(0.toByte(), 255.toByte(), 0.toByte())
        val BLUE = ColorRGBb(0.toByte(), 0.toByte(), 255.toByte())
        val YELLOW = ColorRGBb(255.toByte(), 255.toByte(), 0.toByte())
        val CYAN = ColorRGBb(0.toByte(), 255.toByte(), 255.toByte())
        val MAGENTA = ColorRGBb(255.toByte(), 0.toByte(), 255.toByte())
        val SILVER = ColorRGBb(192.toByte(), 192.toByte(), 192.toByte())
        val GRAY = ColorRGBb(128.toByte(), 128.toByte(), 128.toByte())
        val MAROON = ColorRGBb(128.toByte(), 0.toByte(), 0.toByte())
        val OLIVE = ColorRGBb(128.toByte(), 128.toByte(), 0.toByte())
        val GREEN = ColorRGBb(0.toByte(), 128.toByte(), 0.toByte())
        val PURPLE = ColorRGBb(128.toByte(), 0.toByte(), 128.toByte())
        val TEAL = ColorRGBb(0.toByte(), 128.toByte(), 128.toByte())
        val NAVY = ColorRGBb(0.toByte(), 0.toByte(), 128.toByte())
        val CORNFLOWERBLUE = ColorRGBb(100.toByte(), 149.toByte(), 237.toByte())
    }

}
